docs/maps/index.rst

```diff
@@ -210,14 +216,134 @@ The following demonstrates how one can set pixel values:
    from gammapy.maps import Map
    m = Map.create(binsz=0.1, map_type='wcs', width=10.0)
 
-   m.set_by_coords( ([-0.05,-0.05],[0.05,0.05]), [0.5, 1.5] )
-   m.fill_by_coords( ([-0.05,-0.05],[0.05,0.05]), weights=[0.5, 1.5] )
+   m.set_by_coord( ([-0.05,-0.05],[0.05,0.05]), [0.5, 1.5] )
+   m.fill_by_coord( ([-0.05,-0.05],[0.05,0.05]), weights=[0.5, 1.5] )
+
+Interface with `~MapCoord` and `~astropy.coordinates.SkyCoord`
+--------------------------------------------------------------
+
+The ``coord`` accessor methods accept `dict`, `~MapCoord`, and
+`~astropy.coordinates.SkyCoord` arguments in addition to the standard
+`tuple` of `~numpy.ndarray` argument.  When using a `tuple` argument a
+`~astropy.coordinates.SkyCoord` can be used instead of longitude and
+latitude arrays.  The coordinate frame of the
+`~astropy.coordinates.SkyCoord` will be transformed to match the
+coordinate system of the map.
+
+.. code:: python
+
+   import numpy as np
+   from astropy.coordinates import SkyCoord
+   from gammapy.maps import Map, MapCoord, MapAxis
+
+   lon = np.array([0.0,1.0])
+   lat = np.array([1.0,2.0])
+   energy = np.array([100., 1000.])
+   energy_axis = MapAxis.from_bounds(100., 1E5, 12, interp='log', name='energy')
+   
+   skycoord = SkyCoord(lon, lat, unit='deg', frame='galactic')   
+   m = Map.create(binsz=0.1, map_type='wcs', width=10.0,
+                  coordsys='GAL', axes=[energy_axis])
+
+   m.set_by_coord( (skycoord, energy), [0.5, 1.5] )
+   m.get_by_coord( (skycoord, energy) )
```

Comment:
@woodmd - I've pushed some small fixes in f7e10bfe .

This one maps docs example still errors out and I fail to see why!?
```
In [31]: %paste
   import numpy as np
   from astropy.coordinates import SkyCoord
   from gammapy.maps import Map, MapCoord, MapAxis

   lon = np.array([0.0,1.0])
   lat = np.array([1.0,2.0])
   energy = np.array([100., 1000.])
   energy_axis = MapAxis.from_bounds(100., 1E5, 12, interp='log', name='energy')
   
   skycoord = SkyCoord(lon, lat, unit='deg', frame='galactic')   
   m = Map.create(binsz=0.1, map_type='wcs', width=10.0,
                  coordsys='GAL', axes=[energy_axis])

   m.set_by_coord( (skycoord, energy), [0.5, 1.5] )
   m.get_by_coord( (skycoord, energy) )
## -- End pasted text --
---------------------------------------------------------------------------
KeyError                                  Traceback (most recent call last)
<ipython-input-31-07934e830df0> in <module>()
     13 
     14 m.set_by_coord( (skycoord, energy), [0.5, 1.5] )
---> 15 m.get_by_coord( (skycoord, energy) )

~/code/gammapy/gammapy/maps/base.py in get_by_coord(self, coords)
    485         vals = np.empty(coords.shape, dtype=self.data.dtype)
    486         coords = coords.apply_mask(msk)
--> 487         idx = self.geom.coord_to_idx(coords)
    488         vals[msk] = self.get_by_idx(idx)
    489         vals[~msk] = np.nan

~/code/gammapy/gammapy/maps/geom.py in coord_to_idx(self, coords, clip)
   1045         pix = self.coord_to_pix(coords)
   1046         return self.pix_to_idx(pix, clip=clip)
-> 1047 
   1048     @abc.abstractmethod
   1049     def pix_to_coord(self, pix):

~/code/gammapy/gammapy/maps/wcs.py in coord_to_pix(self, coords)
    486             return tuple([np.array([]) for i in range(coords.ndim)])
    487 
--> 488         c = self.coord_to_tuple(coords)
    489         # Variable Bin Size
    490         if self.axes and self.npix[0].size > 1:

~/code/gammapy/gammapy/maps/geom.py in coord_to_tuple(self, coord)
   1185             coord_tuple += [coord[ax.name]]
   1186 
-> 1187         return coord_tuple
   1188 
   1189     @abc.abstractmethod

~/code/gammapy/gammapy/maps/geom.py in __getitem__(self, key)
    629         if isinstance(key, six.string_types):
    630             return self._data[key]
--> 631         else:
    632             return list(self._data.values())[key]
    633 

KeyError: 'energy'
```

Reply:
This occurs due to a bug in `apply_mask`.  I'm committing a fix now.